rust/chains/tw_polkadot/src/call_encoder/mod.rs

```diff
@@ -0,0 +1,206 @@
+                // TODO: `BondAndNominate` needs 3 call_indices values to support this.
+                //call_indices: ban.call_indices.clone(),
+                call_indices: None,
+            }),
+        }))?;
+
+        // Encode a nominate call
+        let second = self.encode_call(&SigningVariant::staking_call(Proto::Staking {
+            message_oneof: StakingVariant::nominate(Nominate {
+                nominators: ban.nominators.clone(),
+                // TODO: `BondAndNominate` needs 3 call_indices values to support this.
+                //call_indices: ban.call_indices.clone(),
+                call_indices: None,
+            }),
+        }))?;
+
+        // Encode both calls as batched
+        self.encode_batch(vec![first, second], &ban.call_indices)
+    }
+
+    fn encode_staking_chill_and_unbond(&self, cau: &ChillAndUnbond) -> EncodeResult<RawOwned> {
+        let first = self.encode_call(&SigningVariant::staking_call(Proto::Staking {
+            message_oneof: StakingVariant::chill(Chill {
+                // TODO: `ChillAndUnbond` needs 3 call_indices values to support this.
+                //call_indices: cau.call_indices.clone(),
+                call_indices: None,
+            }),
+        }))?;
+
+        let second = self.encode_call(&SigningVariant::staking_call(Proto::Staking {
+            message_oneof: StakingVariant::unbond(Unbond {
+                value: cau.value.clone(),
+                // TODO: `ChillAndUnbond` needs 3 call_indices values to support this.
+                //call_indices: cau.call_indices.clone(),
+                call_indices: None,
+            }),
+        }))?;
+
+        // Encode both calls as batched
+        self.encode_batch(vec![first, second], &cau.call_indices)
+    }
+
+    fn encode_staking_batch_call(&self, s: &Staking) -> EncodeResult<Option<RawOwned>> {
+        match &s.message_oneof {
+            StakingVariant::bond_and_nominate(ban) => {
+                let batch = self.encode_staking_bond_and_nominate(&ban)?;
+                Ok(Some(batch))
+            },
+            StakingVariant::chill_and_unbond(cau) => {
+                let batch = self.encode_staking_chill_and_unbond(&cau)?;
+                Ok(Some(batch))
+            },
+            _ => Ok(None),
+        }
+    }
+
+    pub fn encode_call(&self, msg: &SigningVariant<'_>) -> EncodeResult<RawOwned> {
+        // Special case for batches.
+        match msg {
+            SigningVariant::balance_call(b) => {
+                if let Some(batch) = self.encode_balance_batch_call(b)? {
+                    return Ok(batch);
+                }
+            },
+            SigningVariant::staking_call(s) => {
+                if let Some(batch) = self.encode_staking_batch_call(s)? {
+                    return Ok(batch);
+                }
+            },
+            _ => (),
+        }
+        // non-batch calls.
+        self.encoder.encode_call(msg)
```

Comment:
> I find these batch and non-batch procession suboptimal and difficult to understand.
That is because of how the `Polkadot.proto` has special cases for batching.  Batching should not be handled inside each pallet.  On Substrate networks batching of calls is handled by the `Utility` pallet and each network can have some differences in that pallet.

I have tried multiple times to handle those special case batching inside the pallet (Balance, Staking), but it make all other simple call cases more complex.

In the next PR for Polymesh I have use a more generic batching method that simplifies the call encoding a lot:
Call encoder:
https://github.com/PolymeshAssociation/wallet-core/blob/polymesh_rust_impl/rust/chains/tw_polymesh/src/call_encoder.rs
Polymesh.proto:
https://github.com/PolymeshAssociation/wallet-core/blob/polymesh_rust_impl/src/proto/Polymesh.proto

The same designed could be used for Polkadot, but changing the proto file would break frontend clients.